R/database.R

```diff
@@ -2628,3 +2630,385 @@ get_gene_expression = function(metadata,
+#' This function requires 1 sample specified in the this_sample_id parameter, or a metadata subset to 1 sample (these_samples_metadata). For multiple samples, please refer to get_manta_sv_by_samples.
+#'
+#' @param min_vaf The minimum tumour VAF for a SV to be returned.
+#' @param min_score The lowest Manta somatic score for a SV to be returned.
+#' @param pass If set to TRUE, include SVs that are annotated with PASS in FILTER column. Default is TRUE.
+#' @param this_sample_id The single sample_id you want to obtain the result from.
+#' @param these_samples_metadata A metadata table containing metadata for this_sample_id, or sample of interest. If no sample ID is provided with this_sample_id, the function will extract sample ID from these_samples_metadata.
+#' @param with_chr_prefix Prepend all chromosome names with chr (required by some downstream analyses).
+#' @param from_flatfile Set to TRUE by default.
+#' @param projection The projection genome build.
+#' @param all_sv_grch37 Optional data frame with merged manta results from get_manta_sv(), useful for when individual sample in a specific genome projection is not available.
+#' @param all_sv_hg83 Optional data frame with merged manta results from get_manta_sv(), useful for when individual sample in a specific genome projection is not available.
+#'
+#' @return a data frame containing the Manta outputs from this_sample_id in a bedpe-like format with additional columns extracted from the VCF column.
+#' @import tidyverse
+#' @export
+#'
+#' @examples
+#' #minimum parameters
+#' my_bedpe_df = get_manta_sv_by_sample(this_sample_id = "00-14595_tumorA")
+#'
+#' #extended parameter usage
+#' my_metadata = get_gambl_metadata() %>% dplyr::filter(sample_id = "00-14595_tumorA")
+#'
+#' my_bedpe_df = get_manta_sv_by_sample(these_samples_metadata = my_metadata,
+#'                                      min_vaf = 0.05,
+#'                                      min_score = 30,
+#'                                      pass = FALSE,
+#'                                      with_chr_prefix = TRUE)
+#'
+get_manta_sv_by_sample = function(min_vaf = 0.1,
+                                  min_score = 40,
+                                  pass = TRUE,
+                                  this_sample_id,
+                                  these_samples_metadata,
+                                  with_chr_prefix = FALSE,
+                                  from_flatfile = TRUE,
+                                  projection = "grch37",
+                                  all_sv_grch37,
+                                  all_sv_hg38){
+
+  #check remote configuration
+  remote_session = check_remote_configuration(auto_connect = TRUE)
+
+  #retrieve sample ID from metadata, or get metadata for provided sample ID
+  #get sample ID from metadata and check if metadata contains more than 1 row
```

Comment:
This is a very convoluted approach that will cause this function to be called a bunch of times unnecessarily. Extra overhead that can be avoided. See my comments on the function that calls this one.

Reply:
is this resolved?